src/Explain.h

```diff
@@ -32,199 +36,206 @@
+            if (command.size() == 2) {
+                query = parseTuple(command[1]);
+            } else {
+                printError("Usage: explain relation_name(\"<string element1>\", <number element2>, ...)\n");
+                return true;
+            }
+            std::unique_ptr<TreeNode> t =
+                    prov.explain(query.first, query.second, ExplainConfig::getExplainConfig().depthLimit);
+            printTree(std::move(t));
+        } else if (command[0] == "subproof") {
+            std::pair<std::string, std::vector<std::string>> query;
+            int label = -1;
+            if (command.size() > 1) {
+                query = parseTuple(command[1]);
+                label = std::stoi(query.second[0]);
+            } else {
+                printError("Usage: subproof relation_name(<label>)\n");
+                return true;
+            }
+            std::unique_ptr<TreeNode> t =
+                    prov.explainSubproof(query.first, label, ExplainConfig::getExplainConfig().depthLimit);
+            printTree(std::move(t));
+        } else if (command[0] == "explainnegation") {
+            std::pair<std::string, std::vector<std::string>> query;
+            if (command.size() == 2) {
+                query = parseTuple(command[1]);
+            } else {
+                printError(
+                        "Usage: explainnegation relation_name(\"<string element1>\", <number element2>, "
+                        "...)\n");
+                return true;
+            }
 
-                if (ncurses && !output) {
-                    prefresh(treePad, 0, 0, 0, 0, maxy - 3, maxx - 1);
-                }
+            size_t i = 1;
+            std::string rules;
+            for (auto rule : prov.getRules(query.first)) {
+                rules += std::to_string(i) + ": ";
+                rules += rule;
+                rules += "\n\n";
+                i++;
+            }
+            printInfo(rules);
 
-                printPrompt("Pick a rule number: ");
+            printPrompt("Pick a rule number: ");
 
-                std::string ruleNum = getInput();
-                auto variables =
-                        prov.explainNegationGetVariables(query.first, query.second, std::stoi(ruleNum));
+            std::string ruleNum = getInput();
+            auto variables = prov.explainNegationGetVariables(query.first, query.second, std::stoi(ruleNum));
 
-                if (variables.size() == 1 && variables[0] == "@") {
-                    printPrompt("The tuple exists, cannot explain negation of it!\n");
-                    continue;
-                } else if (variables.size() == 1 && variables[0] == "@non_matching") {
-                    printPrompt("The variable bindings don't match, cannot explain!\n");
-                    continue;
-                }
+            if (variables.size() == 1 && variables[0] == "@") {
+                printInfo("The tuple exists, cannot explain negation of it!\n");
```

Comment:
Would it be better to say, "Cannot explain negations"? If so, use something like that.